e2e/container/create_test.go

```diff
@@ -0,0 +1,80 @@
+package container
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/docker/cli/e2e/internal/fixtures"
+	"github.com/gotestyourself/gotestyourself/icmd"
+)
+
+func TestCreateWithContentTrust(t *testing.T) {
+	dir := fixtures.SetupConfigFile(t)
+	defer dir.Remove()
+	image := fixtures.CreateMaskedTrustedRemoteImage(t, registryPrefix, "trust-create", "latest")
+
+	result := icmd.RunCmd(
+		icmd.Command("docker", "create", image),
+		fixtures.WithConfig(dir.Path()),
+		fixtures.WithTrust,
+		fixtures.WithNotary,
+	)
+	result.Assert(t, icmd.Expected{
+		Err: fmt.Sprintf("Tagging %s@sha", image[:len(image)-7]),
+	})
+}
+
+func TestCreateWithContentTrustEvilNotaryServer(t *testing.T) {
+	dir := fixtures.SetupConfigFile(t)
+	defer dir.Remove()
+	evilDir := fixtures.SetupConfigFile(t)
+	defer evilDir.Remove()
+	image := fixtures.CreateMaskedTrustedRemoteImage(t, registryPrefix, "trust-create", "latest")
+
+	t.Log(dir, evilDir)
+
+	result := icmd.RunCmd(
+		icmd.Command("docker", "create", image),
+		fixtures.WithConfig(dir.Path()),
+		fixtures.WithTrust,
+		fixtures.WithNotary,
+	)
+	result.Assert(t, icmd.Expected{
+		Err: fmt.Sprintf("Tagging %s@sha", image[:len(image)-7]),
+	})
+
+	icmd.RunCommand("docker", "image", "rm", image)
+
+	// In order to make an evil server, lets re-init a client (with a different trust dir) and push new data.
+	// tag an image and upload it to the private registry
+	// and push to the evil server
+	icmd.RunCmd(
+		icmd.Command("docker", "tag", fixtures.AlpineImage, image),
+		fixtures.WithConfig(evilDir.Path()),
+	).Assert(t, icmd.Success)
+	result = icmd.RunCmd(
+		icmd.Command("docker", "push", image),
+		fixtures.WithConfig(evilDir.Path()),
+		fixtures.WithTrust,
+		fixtures.WithNotaryServer(fixtures.EvilNotaryURL),
+		fixtures.WithPassphrase("foo", "bar"),
+	)
+	result.Assert(t, icmd.Expected{
+		Out: "Signing and pushing trust metadata",
+	})
+
+	// FIXME(vdemeester) doesn't work
+	/*
+		// Now, try running with the original client from this new trust server. This should fail because the new root is invalid.
```

Comment:
@endophage @n4ss what's the expected behavior here ? (the commented code comes from initial moby test suite, but fails on current `docker/cli`)